Library/Homebrew/download_strategy.rb

```diff
@@ -576,13 +576,17 @@ def externals
   end
 
   def fetch_repo(target, url, revision = nil, ignore_externals = false)
-    # Use "svn up" when the repository already exists locally.
+    # Use "svn update" when the repository already exists locally.
     # This saves on bandwidth and will have a similar effect to verifying the
     # cache as it will make any changes to get the right revision.
-    svncommand = target.directory? ? "up" : "checkout"
+    svncommand = target.directory? ? "update" : "checkout"
     args = ["svn", svncommand]
-    args << url unless target.directory?
-    args << target
+    if svncommand == "checkout"
+      args << url
+      args << target
+    elsif svncommand == "update"
+      args << svn_escape(target)
+    end
```

Comment:
It's still checking for the same thing twice. Make this

```ruby
args = if target.directory?
  ["svn", "update", svn_escape(target)]
else
  ["svn", "checkout", url, target]
end
```

That make me think: Is `svn update target` the same as `cd target; svn update`?

Reply:
(args)
Oh, that's indeed way better. I didn't see it and I'm blaming my lack of ruby skills! ;-)

Reply:
@reitermarkus , about the ```cd target```

Yes, the effect of ```cd target@peg; svn update``` is the same as ```svn update target@peg@``` - to the repository, at least. 

I thought of that but as I was not sure if or how brew could get affected to these path changes, I kept with this "maybe more conservative approach". As per your comment I looked a bit further and decided to give it a try.

So the export command ended up like this and worked fine (thanks to the ```chdir```).
```
      system_command! "svn", 
                      args: ["export", "--force", ".", unpack_dir],
                      chdir: path.to_s
```

The update looked like this and it worked (note: no output is read from it).
```
    args = if target.directory?
      ["cd", target.to_s, ";", "svn", "update"]
    else
      ["svn", "checkout", url, target]
    end
```

However, I could not make the ```svn info``` work with ```popen_read```. It's used by the methods ```last_commit```, ```repo_url``` and ```source_modified_time```. Looks like ```popen_read``` keeps returning the output of the first command (cd).

Tried this:
```
    xml = REXML::Document.new(Utils.popen_read("cd", cached_location.to_s, ";", "svn", "info", "--xml"))
```
and even tried with ```&> /dev/null```:
```
    xml = REXML::Document.new(Utils.popen_read("cd", cached_location.to_s, "&> /dev/null;", "svn", "info", "--xml"))
```

In the end, even if we manage to get past this problem in ```svn info```, I think the svn_escape() is easier to read and may be easier to understand in the future.

One option would be bringing svn_escape() back to ```download_strategy``` and using it only there, leaving the export command with the chdir.

What do you think?

Reply:
Instead of `popen_read` you can also use `system_command`. It returns a result which you can call `stdout` on.

Reply:
I didn't test with ```system_command``` yet, but do you think this is better?
```
# with svn_escape
xml = REXML::Document.new(Utils.popen_read( "svn", "info", "--xml", svn_escape(cached_location)))
...
# so far, with system_command, cd ..; & .stdout
xml = REXML::Document.new(system_command("cd", cached_location.to_s, ";", "svn", "info", "--xml").stdout)
```
To me it seems less obscure using ```svn_escape```... and there's the possibility that more adjustments might have to be done (as it didn't work already with ```popen_read```).

Reply:
```ruby
info = system_command("svn", "info", "--xml", chdir: cached_location.to_s).stdout
xml = REXML::Document.new(info)
```